packages/sdk/src/content_topic.ts

```diff
@@ -0,0 +1,110 @@
+import type { Multiaddr } from "@multiformats/multiaddr";
+import {
+  createDecoder,
+  DecodedMessage,
+  waitForRemotePeer,
+  WakuNode
+} from "@waku/core";
+import {
+  Callback,
+  IDecoder,
+  IFilterSubscription,
+  IMetadata,
+  Protocols
+} from "@waku/interfaces";
+import {
+  contentTopicToPubsubTopic,
+  shardInfoToPubsubTopics
+} from "@waku/utils";
+
+import { createLightNode } from ".";
+
+// Given a Waku node, peer Multiaddr, and content topic, creates a decoder and
+// subscription for that content topic.
+export async function prepareSubscription(
+  waku: WakuNode,
+  contentTopic: string,
+  peer?: Multiaddr
+): Promise<{
+  decoder: IDecoder<DecodedMessage>;
+  subscription: IFilterSubscription;
+}> {
+  // Validate that the Waku node matches assumptions
+  if (!waku.filter) {
+    throw new Error("Filter protocol missing from Waku node");
+  }
+  const shardInfo = (waku.libp2p.components.metadata as IMetadata).shardInfo;
+  if (!shardInfo) {
+    throw new Error("Shard info missing from Waku node.");
+  }
+
+  // Validate content topic and ensure node is configured for its corresponding pubsub topic
+  const pubsubTopics = shardInfoToPubsubTopics(shardInfo);
+  const pubsubTopic = contentTopicToPubsubTopic(contentTopic);
+  if (!pubsubTopics.includes(pubsubTopic))
+    throw new Error(
+      "Content topic does not match any pubsub topic in shard info."
+    );
+
+  // Ensures there is a peer. Without this condition, the subscription will fail to create.
+  if (peer) {
+    await waku.dial(peer);
+    await waitForRemotePeer(waku, [Protocols.Filter]);
+  }
+
+  // Create decoder and subscription
+  const decoder = createDecoder(contentTopic, pubsubTopic);
+  const subscription = await waku.filter.createSubscription(pubsubTopic);
+
+  return { decoder, subscription };
+}
+
+export async function streamContentTopic(
+  contentTopic: string,
+  waku?: WakuNode,
+  peer?: Multiaddr
+): Promise<[ReadableStream<DecodedMessage>, WakuNode]> {
+  waku =
+    waku ??
+    ((await createLightNode({
+      shardInfo: { contentTopics: [contentTopic] }
+    })) as WakuNode);
```

Comment:
why are we typecasting the type `LightNode` as `WakuNode`? if there's some API we require, we can add it part of the `LightNode` interface if necessary

Reply:
replaced with `LightNode` in last commit